src/components/NcEmptyContent/NcEmptyContent.vue

```diff
@@ -121,57 +121,73 @@ export default {
 ```
 </docs>
 
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+withDefaults(defineProps<{
+	/**
+	 * A header message about an empty content shown.
+	 *
+	 * @example 'No comments'
+	 */
+	name?: string
+
+	/**
+	 * Desription of the empty content
+	 *
+	 * @example 'No comments yet, start the conversation!'
+	 */
+	description?: string
+}>(), {
+	description: '',
+	name: '',
+})
+
+defineSlots<{
+	/**
+	 * Optional slot for a button or the like
+	 */
+	action?: Slot
+
+	/**
+	 * Optional slot for adding an icon
+	 */
+	icon?: Slot
+
+	/**
+	 * Allow to add custom formatted name as an alternative to the name property.
+	 * The content passed shall be enclosed by a header element.
+	 */
+	name?: Slot
+
+	/**
+	 * Optional formatted description rendered inside a paragraph as an alternative to the description property.
+	 */
+	description?: Slot
+}>()
+</script>
+
 <template>
 	<div class="empty-content" role="note">
-		<div v-if="$slots.icon" class="empty-content__icon" aria-hidden="true">
-			<!-- @slot Optional material design icon -->
+		<div class="empty-content__icon" aria-hidden="true">
```

Comment:
To keep previous behavior and not render DOM nodes when they are not used

```suggestion
		<div v-if="$slots.icon" class="empty-content__icon" aria-hidden="true">
```

Reply:
But then we again need to check availability of slots?

Reply:
What do you mean? These `v-if`s did it...